Packs/WootCloud/Integrations/WootCloud/WootCloud.py

```diff
@@ -0,0 +1,255 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+''' IMPORTS '''
+
+import json
+import requests
+from datetime import datetime
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' GLOBALS/PARAMS '''
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+USERNAME = demisto.params().get('client_id')
+PASSWORD = demisto.params().get('secret_key')
+SERVER = "https://api.wootuno.wootcloud.com"
+SSL_VERIFY = True
+# Service base URL
+API_VERSION = '/v1/'
+# Headers to be sent in requests
+HEADERS = {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json',
+    'Host': 'api.wootuno.wootcloud.com'
+}
+# How much time to begin retrieving incidents before the first fetch
+FETCH_TIME = demisto.params().get('fetch_time', '3 days')
+# Fetch filters
+ALERT_TYPE = demisto.params().get('alert_type')  # Bluetooth, Packet, Anomaly
+SEVERITY_TYPE = demisto.params().get('severity_type')  # Warning, Critical, Notice, Info
+
+
+''' HELPER FUNCTIONS '''
+
+
+def item_to_incident(item):
+    incident = {}
+    # Incident Title
+    incident['name'] = 'Incident: ' + item.get('id')
+    # Incident occurrence time, usually item creation date in service
+    incident['timestamp'] = item.get('timestamp')
+    # The raw response from the service, providing full info regarding the item
+    incident['rawJSON'] = json.dumps(item)
+    return incident
+
+
+def iter_all_alerts(client, type, start, end, severity=None, skip=None, limit=10, site_id=None):
+    """
+    Iterate through packet, bluetooth, or anomaly alerts generated in requested time span.
+    """
+    first = client.get_woot_alerts(type, start, end, severity, skip, limit, site_id)
+
+    alert_type = ''
+    if type == 'packet':
+        alert_type = 'packet_alerts'
+    elif type == 'bluetooth':
+        alert_type = 'alerts'  # NOTE: bluetooth alert has same as anamoly alert
+    elif type == 'anomaly':
+        alert_type = 'alerts'
+
+    total = first['total']
+    alerts = first[alert_type]
+    step = limit
+    while step < total:
+        alerts += client.get_woot_alerts(type, start, end, severity, step, limit, site_id)[alert_type]
+        step += limit
+    return alerts
+
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """
+    Client will implement the service API, and should not contain any Demisto logic.
+    Should only do requests and return data.
+    """
+    def __init__(self, server, ssl_verify, api_version, headers, username, password):
+        self.server = server
+        self.ssl_verify = ssl_verify
+        self.api_version = api_version
+        self.headers = headers
+        self.username = username
+        self.password = password
+
+    def http_request(self, method, url_suffix, params=None, json=None):
```

Comment:
Better use the http_request from the base client in common server python script. 
We are handling all error and all arguments there.

Reply:
commited

Reply:
committed